packages/grafana-data/src/dataframe/DataFrameJSON.test.ts

```diff
@@ -0,0 +1,81 @@
+import { FieldType } from '../types/dataFrame';
+import { DataFrameJSON, dataFrameFromJSON } from './DataFrameJSON';
+
+describe('DataFrame JSON', () => {
+  describe('when called with a DataFrame', () => {
+    it('then it should reverse the order of values in all fields', () => {
+      const json: DataFrameJSON = {
+        schema: {
+          fields: [
+            { name: 'time', type: FieldType.time },
+            { name: 'name', type: FieldType.string },
+            { name: 'value', type: FieldType.number },
+          ],
+        },
+        data: [
+          [100, 200, 300],
+          ['a', 'b', 'c'],
+          [1, 2, 3],
+        ],
+        replaced: [
+          {}, // nothing to replace, but keeps the index
```

Comment:
could even be `null` to skip object allocation/encoding ;)

Reply:
this started out without the fill, and the perf difference was quite large:

https://github.com/leeoniya/uPlot/blob/6ad1cb86e85672604515045abca3c891aab94086/src/utils.js#L377

Reply:
:+1: